packages/shared/src/hooks/notifications/useEnableNotification.ts

```diff
@@ -49,16 +57,22 @@ export const useEnableNotification = ({
     [source, onEnablePush],
   );
 
+  const onToggle = useCallback(
```

Comment:
Why did we have to add this function? Is it to immediately change the user's preference? If so, we should make it similar to the existing implementation where we only change it once the user submits.

Reply:
Also, once the user submits having the toggled on, we should not show it anymore.

Reply:
Exactly. I think for the case of editing an existing comment, it does not make sense to wait for the submit of comment.

Also, the existing function does not unsubscribe on dismiss as far as I understand. Which is what the toggle needs to achieve.

Let's sync on this together to figure out what the actual behavior should be.